gu-provider/src/main.rs

```diff
@@ -43,16 +42,19 @@ mod sync_exec;
 const VERSION: &str = env!("CARGO_PKG_VERSION");
 
 fn main() {
+    let config_module = gu_persist::config::ConfigModule::new();
+
     GuApp(|| App::new("Golem Unlimited Provider").version(VERSION)).run(
         LogModule
-            .chain(AutocompleteModule::new())
-            .chain(gu_persist::config::ConfigModule::new())
+            .chain(daemon_module::DaemonModule::provider(
+                config_module.work_dir().to_path_buf(),
+            )).chain(server::ServerModule::new())
+            .chain(config_module)
             .chain(gu_lan::module::LanModule::module())
             .chain(gu_hardware::module())
             .chain(status::module())
             .chain(connect::module())
-            .chain(daemon_module::DaemonModule::provider())
             .chain(permission::module())
-            .chain(server::ServerModule::new()),
+            .chain(AutocompleteModule::new()),
```

Comment:
Now chain doesn't look pretty. I think extracting from a decorator is better.
But it does not matter cause daemon is going to provide library functions instead of Module implementation.